lib/azure/armrest/model/storage_account.rb

```diff
@@ -194,27 +193,43 @@ def directory_metadata(share, directory, key = access_key, options = {})
       # * marker
       # * maxresults
       # * timeout
+      # * all
+      #
+      # If the :all option is set to true, then this method will collect all
+      # records. Otherwise, it is capped at 5000 records by Azure, or whatever
+      # you set :maxresults to. If you set both the :all option and the :maxresults
+      # option, then all records will be collected in :maxresults batches.
       #
       def files(share, key = access_key, options = {})
         raise ArgumentError, "No access key specified" unless key
 
-        query = {:restype => 'directory', :comp => 'list'}.merge(options).to_query
+        query = "comp=list&restype=directory"
+
+        skip_defs = options[:skip_accessors_definition]
+
+        options.each do |okey, ovalue|
+          unless okey == :skip_accessors_definition
+            query += "&#{okey}=#{[ovalue].flatten.join(',')}"
+          end
+        end
 
         response = file_response(key, query, 'get', nil, share)
 
-        doc = Nokogiri::XML(response.body)
+        hash = Hash.from_xml(response.body)['EnumerationResults']['Entries']
```

Comment:
Can it ever fail? Does `['EnumerationResults']` alway present?

Reply:
In all my tests, yes, it's always present.